Updating Solid/Constitutive/GPfem/Umat

josep-m-carbonell opened the pull request:
1. new ConstitutiveModelsApplication (must be set ON for compilation of the SolidMechanics)
2. new UmatApplication (for geotechnical models)
3. update and reorganization of the SolidMechanicsApplication
4. update of the Pfem features 
5. submodule for the tetgen
6. needed compatibility with the GiDInteface for the SolidMechanicsApp and PfemApp
7. small changes in depending applications (must be checked for the responsible developers)

lluis-mv committed:
First commit

Remove lots of functions, change the base class of Umat, register constitutive, change the CalculateMaterial interface, etc

lluis-mv committed:
Small test. Somehow works but...

It somehow works, the but i have problems with:
^ The constitutive equation requieres several steps to converge (altough its a small strains linear problem)
^ I have problems with the Materials.py. It says it can not be cloned. I do not understand it and I just don't clone. (the problem is elsewhere because SolidMechanics constitutive equations can not be cloned). Big trouble

lluis-mv committed:
now it works

the fortran is an hypoelastic-plastic. The solution was "finalized" at every iteration, and... 
Now it works, however several problems remain:
* introduction of constitutive models
* other than 3D cases,...

lluis-mv committed:
Typos warnings and appropiate Clone

lluis-mv committed:
clone in python

now, it clones correctly in python. (I just copied the SolidMechanics includes to here...)

lluis-mv committed:
Changes in umat EECC

Modifying what is saved in the umat law
Adding a pseudo-large strains version with hypoelasticity and lie 
derivative (I thinck is correct).

lluis-mv committed:
water pressure utility. Stabilization

the stabilization parameter was not completely correct

lluis-mv committed:
adding missing term in MCC constitutive matrix

description

lluis-mv committed:
BeginImplementing J2 SS

Begin implementing J2 for SS. For the moment, no changes are introduced in non_linear_assosiatvei... since the same procedures may be used to compute the CauchyStressTensor (the TangentMatrix, Update StrainTensor are not yet included). Further changes are needed to compute the trial state (for the moment I begin from everything zero. Do not FinalizeSolutionStep since will create strange behaviours.

josep-m-carbonell committed:
hotfix-explicit-solver

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
Merge branch 'master' into feature-small-strains-laws

josep-m-carbonell committed:
Merge branch 'feature-small-strains-laws' into develop

josep-m-carbonell committed:
axisym to axisymmetric

simplify loads and use the complete name for axisymmetric

josep-m-carbonell committed:
solve warnings pfem BigText

josep-m-carbonell committed:
Merge branch 'master' into solid_mechanics_develop

josep-m-carbonell committed:
overrides added

to solve warnings in clang compilation

josep-m-carbonell committed:
add overrides to contact mechanics app

to solver clang warnings

josep-m-carbonell committed:
add overrides to pfem solid app

to solve clang warnings

josep-m-carbonell committed:
avoid dependence of the pfem solid and constitutive models app

josep-m-carbonell committed:
comment the clang warning to not override

josep-m-carbonell committed:
fix a warning of a non return method

josep-m-carbonell committed:
updating the conditions structure

josep-m-carbonell committed:
Merge branch 'master' into feature-hash-variables-list

josep-m-carbonell committed:
Merge branch 'feature-hash-variables-list' into solid_mechanics_develop_hash

josep-m-carbonell committed:
Merge branch 'solid_mechanics_develop_hash' into solid_mechanics_develop

josep-m-carbonell committed:
deletetion of a utilities not needed methods

josep-m-carbonell committed:
adapt beam elements

josep-m-carbonell committed:
redistribution of the elements by type

josep-m-carbonell committed:
hotfix for the AddDofs

when dofs are added qhen the mdpa is read, the faster methodology to add dofs fails, so it can not be used.

josep-m-carbonell committed:
clean some printing

josep-m-carbonell committed:
starting conditions re-structure

josep-m-carbonell committed:
starting moment conditions

josep-m-carbonell committed:
add new conditions

ready to test

josep-m-carbonell committed:
update loops

josep-m-carbonell committed:
starting beam base object

josep-m-carbonell committed:
update beam element

josep-m-carbonell committed:
fix duplicated name from a previous pull request

some python files from the solid mechanics were duplicated in python files of the core without correcting them were merged to master in a

josep-m-carbonell committed:
Merge branch 'beam-develop' into solid_mechanics_develop

josep-m-carbonell committed:
correction in the beam utils

josep-m-carbonell committed:
starting small displacement beam

theory to determine...

josep-m-carbonell committed:
change E3 to E1

josep-m-carbonell committed:
new small displacement beam

josep-m-carbonell committed:
deriving large displacements beams from the base

some noalias added too

josep-m-carbonell committed:
change pow by a product

josep-m-carbonell committed:
delete definition of M_PI

josep-m-carbonell committed:
update utilities with template vector

josep-m-carbonell committed:
update beam elemens and schemes

josep-m-carbonell committed:
update a header from mpetracca

josep-m-carbonell committed:
update explicit strategy and python solver files

josep-m-carbonell committed:
add DIMENSION to python

josep-m-carbonell committed:
domain_size to dimension CM

josep-m-carbonell committed:
domain_size to dimension Pfem

josep-m-carbonell committed:
domain_size to dimension PfemFluid

josep-m-carbonell committed:
domain_size to dimension PfemSolid

josep-m-carbonell committed:
domain_size to dimension Solid

and solvers corrections

josep-m-carbonell committed:
correction in local axes

josep-m-carbonell committed:
Merge branch 'master' into solid_mechanics_develop

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
correction in volume force

josep-m-carbonell committed:
Merge branch 'migration-modal-analysis' into solid_mechanics_develop

josep-m-carbonell committed:
correction after merging migration branch

josep-m-carbonell committed:
fix spaces and position of new methods

a typedef was also needed

josep-m-carbonell committed:
small correction in beam

josep-m-carbonell committed:
small correction in tests

josep-m-carbonell committed:
corrections to run tests correctly

a python utility was missing

josep-m-carbonell committed:
corrections in tests and beam load

josep-m-carbonell committed:
Merge branch 'master' into solid_mechanics_develop

josep-m-carbonell committed:
Merge remote-tracking branch 'origin/solid_mechanics_develop' into solid_mechanics_develop

josep-m-carbonell committed:
Merge branch 'feature-printing-eigenvalues' into solid_mechanics_develop

josep-m-carbonell committed:
solve compatibility with the new parameter in eigensolver

lluis-mv committed:
Changes UMAT FILE

changes in the umat (to be consistent with Kirchhoff....).
More changes are required to be "completely correct"

lluis-mv committed:
Changes UMAT FILE

changes in the umat (to be consistent with Kirchhoff....).
More changes are required to be "completely correct"

lluis-mv committed:
beggining SmallStrains J2 from Large strains

first steps of the implementation of J2 at small strains from the Large strains version.
More work is needed.

lluis-mv committed:
beggining SmallStrains J2 from Large strains

first steps of the implementation of J2 at small strains from the Large strains version.
More work is needed.

lluis-mv committed:
Modifications in the ElastoPlastic contact analogy

New PS and Axisym version, correction of erros. Introduce the Parametric Surface discretization in the bounding boxes.

lluis-mv committed:
Modifications in the ElastoPlastic contact analogy

New PS and Axisym version, correction of erros. Introduce the Parametric Surface discretization in the bounding boxes.

lluis-mv committed:
Old ConstitiveLaws: corrections of errors

Correction of errors of the constitutive Laws implemented in PFEM/SMA. The ones that do not work are not compiled but in the nearer future should be removed.

lluis-mv committed:
Old ConstitiveLaws: corrections of errors

Correction of errors of the constitutive Laws implemented in PFEM/SMA. The ones that do not work are not compiled but in the nearer future should be removed.

lluis-mv committed:
typo?

lluis-mv committed:
typo?

lluis-mv committed:
start thinking 3D. VERY LOCAL CHANGES

modifications in the TETGEN code.

lluis-mv committed:
start thinking 3D. VERY LOCAL CHANGES

modifications in the TETGEN code.

lluis-mv committed:
Merge remote-tracking branch 'origin/feature-umat-constitutive' into feature-umat-constitutive

josep-m-carbonell committed:
create new umat application

lluis-mv committed:
Merge branch 'solid_mechanics_develop' into feature-umat-constitutive

lluis-mv committed:
Thickness in axisymmetric elements

Thickness in axisymmetric elements was not set//checked and created nans

lluis-mv committed:
sort of typos

I need to think about the script.py and solver.InitializeStrategy()

lluis-mv committed:
Comment InitializeStrategy?

lluis-mv committed:
Changes in derived in Python solver

I created a "solver" to set the Jacobian Dofs not in the Solid MechanicsApplication but in the PfemSolidMechanics. Then there was some trouble

josep-m-carbonell committed:
Merge branch 'master' into solid_mechanics_develop

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
solve clang compilation issue

josep-m-carbonell committed:
Merge branch 'create_umat_application' into solid_mechanics_develop

josep-m-carbonell committed:
corrections for a proper compilation of the UmatApplication

josep-m-carbonell committed:
corrections for a proper compilation of the UmatApplication

lluis-mv committed:
Set initial stress state

Re-introduction for setting the initial state again (only works for the old constitutive equations)

josep-m-carbonell committed:
changing DeformationGradientF and DeformationGradientF0 names

josep-m-carbonell committed:
changing DeformationGradientF and DeformationGradientF0 names

josep-m-carbonell committed:
bbar element and linear elastic plastiticy

lluis-mv committed:
Merge remote-tracking branch 'origin/create_umat_application' into feature-umat-constitutive

lluis-mv committed:
merge

trying to merge again

josep-m-carbonell committed:
updating sections with json input

josep-m-carbonell committed:
some comments changed

author/co-author distribution

josep-m-carbonell committed:
some comments changed

author/co-author distribution

josep-m-carbonell committed:
Merge branch 'master' into solid_mechanics_develop

josep-m-carbonell committed:
Revert "Merge branch 'feature-umat-constitutive' into develop"

This reverts commit 55494921b244413a84b772abb27d960b48adbc1b.

josep-m-carbonell committed:
Merge branch 'create_umat_application' into develop

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
Merge branch 'develop' into feature-umat-constitutive

josep-m-carbonell committed:
update of the damage in construction

josep-m-carbonell committed:
add damage model

josep-m-carbonell committed:
damage laws ready to test

lluis-mv committed:
Missing endif in problem_type

And introducing new elements and so on to the custom_problem_type of Pfem/SMA

lluis-mv committed:
I destroyed something in the umat?

lluis-mv committed:
I created a problem, that now is solved in a umat file

josep-m-carbonell committed:
Merge branch 'fix-condition-processes-naming' into solid_mechanics_develop

josep-m-carbonell committed:
last merge eliminated a file, added here again

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
Merge branch 'master' into solid_mechanics_develop

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
Merge branch 'master' into solid_mechanics_develop

josep-m-carbonell committed:
Merge branch 'master' into solid_mechanics_develop

josep-m-carbonell committed:
corrections if the pfem fluid to run from interface

josep-m-carbonell committed:
Merge branch 'pfem-fluid-update' into solid_mechanics_develop

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
solution for CalculateValue implemented

compiled but not tested yet

josep-m-carbonell committed:
correction of a segmentation fault

josep-m-carbonell committed:
Merge branch 'pfem-fluid-update' into solid_mechanics_develop

josep-m-carbonell committed:
Merge branch 'feature-umat-constitutive' into solid_mechanics_develop

josep-m-carbonell committed:
Merge branch 'master' into solid_mechanics_develop

lluis-mv committed:
merge

lluis-mv committed:
Merge remote-tracking branch 'origin/solid_mechanics_develop' into feature-umat-constitutive

josep-m-carbonell committed:
printing rigid walls

josep-m-carbonell committed:
printing rigid walls

josep-m-carbonell committed:
tresca class define correction

lluis-mv committed:
Typos

lluis-mv committed:
Small strain von mises

seems to work (at least it compiles)

lluis-mv committed:
Better

I should reconsider some pointers.... blah blah

lluis-mv committed:
with less pointers

change pointers to doubles to VectorType... blah blah

josep-m-carbonell committed:
MeshId away and New checks in some elements and conditions

josep-m-carbonell committed:
Merge branch 'master' into solid_mechanics_develop

josep-m-carbonell committed:
Merge branch 'feature-umat-constitutive' into solid_mechanics_develop

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
correction in adaptive time if walls are written

josep-m-carbonell committed:
correction in adaptive time if walls are written

lluis-mv committed:
Correct how to write surface loads

The normal should point somewhere and it was done in the inverse order

josep-m-carbonell committed:
Merge branch 'master' into solid_mechanics_develop

josep-m-carbonell committed:
point template still not merged

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
automatic solvers

default values for pressure and velocity solver in json parameters

josep-m-carbonell committed:
Merge branch 'pfem-fluid-update' into solid-mechanics-develop

lluis-mv committed:
Change Name GeneralVariables -> ConditionVariables

And Add override

josep-m-carbonell committed:
Merge branch 'cleanup-point-template' into solid-mechanics-develop

josep-m-carbonell committed:
point<3> to point and clarify geometry name

josep-m-carbonell committed:
reorder cmake list

josep-m-carbonell committed:
comment correction

josep-m-carbonell committed:
remove tetgen and triangle

josep-m-carbonell committed:
remove tetgen and triangle

josep-m-carbonell committed:
adding tetgen and triangle as submodules

josep-m-carbonell committed:
Updated submodule Kratos/applications/PfemApplication/external_modules/tetgen

josep-m-carbonell committed:
Removed submodule Kratos/applications/PfemApplication/external_modules/tetgen

josep-m-carbonell committed:
Removed submodule Kratos/applications/PfemApplication/external_modules/triangle

josep-m-carbonell committed:
Added submodule applications/PfemApplication/external_modules/tetgen

josep-m-carbonell committed:
Added submodule applications/PfemApplication/external_modules/triangle

josep-m-carbonell committed:
Updated submodule applications/PfemApplication/external_modules/tetgen

lluis-mv committed:
Merge remote-tracking branch 'origin/develop' into feature-umat-constitutive

lluis-mv committed:
Merge remote-tracking branch 'origin/solid-mechanics-develop' into feature-umat-constitutive

josep-m-carbonell committed:
Updated submodule applications/PfemApplication/external_modules/triangle

josep-m-carbonell committed:
cmakes with modules

josep-m-carbonell committed:
Merge branch 'develop' into solid-mechanics-with-mesh-modules

lluis-mv committed:
First steps to u-w-element

At least it compiles :'P

lluis-mv committed:
It compiles v2

jeje

lluis-mv committed:
First steps to do a strategy suitable for the U-W element

compiles but, ¿it works?

lluis-mv committed:
adding some matrices

more matrices

lluis-mv committed:
Create a pfem_dynamic python solver

The code compiles and may be run without segmentation fault but it does not work yet.
More work is required in the GiD Interface (json)

lluis-mv committed:
GiD INterface modifications for U_W

the dirichlet boundar conditions are not correct yet

lluis-mv committed:
I think that now the interface for dirichlet conditions work

I think, jeje

pedro-navas committed:
Reminder

Reminder of porosity

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
Merge branch 'develop' into solid-mechanics-develop

josep-m-carbonell committed:
Merge branch 'solid-mechanics-develop' into solid-mechanics-with-mesh-modules

pedro-navas committed:
Correct elementary  K matrix

lluis-mv committed:
write gauss point water pressure

lluis-mv committed:
remove error due to _REACTION _ DISPLACEMENT etc (NOT CORRECTLY DONE)

lluis-mv committed:
to me, it does not make sense with the current way to impose dirichlet

lluis-mv committed:
Upsi, I forgot to deallocate allocated memory through pointers

Improve copy constructor. Serializer save, etc.

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
Merge branch 'master' into solid-mechanics-develop

josep-m-carbonell committed:
Merge branch 'develop' into solid-mechanics-develop

josep-m-carbonell committed:
Merge branch 'solid-mechanics-develop' into solid-mechanics-with-mesh-modules

josep-m-carbonell committed:
correction after the change of the eigen solver

skyline_lu to complex_skyline_lu_solver

josep-m-carbonell committed:
correction after the change of the eigen solver

skyline_lu to complex_skyline_lu_solver

lluis-mv committed:
Introducing Hypoplastic clay model (Masin )

and some minor modifications

lluis-mv committed:
Removing constitutive parameters set by hand

Now they go by GetProperties. And also have been included in the CHECK!

lluis-mv committed:
correction. dofs_per_node, get displ---

lluis-mv committed:
Small changes, at least now it works for one-phase.

OK

lluis-mv committed:
More matrices, less matrices

lluis-mv committed:
Finally the signs and everything seems to be correct.

YUPI!

lluis-mv committed:
Introduce variables, remove deprecated variables

Introduce Water_disp/Vel/acc_REACTION, REMOVE IMPOSED_WATER_PRESSURE.
ResidualBasedStaticUwP is not used. Considere removing it.

lluis-mv committed:
Continue with the implementation

lluis-mv committed:
UPS!

josep-m-carbonell committed:
Merge branch 'master' into solid-mechanics-with-mesh-modules

josep-m-carbonell committed:
Merge branch 'solid-mechanics-develop' into solid-mechanics-with-mesh-modules

josep-m-carbonell committed:
Merge branch 'master' into solid-mechanics-with-mesh-modules

josep-m-carbonell committed:
some corrections in the location of triangle and tetgen

josep-m-carbonell committed:
restore triangle in external libraries for cmake kratos core

josep-m-carbonell committed:
trick to solve stabilization factor null value

josep-m-carbonell committed:
adding rotation dofs to generalize conditions for beams and shells

josep-m-carbonell committed:
subfolders to order boundary conditions

josep-m-carbonell committed:
Merge branch 'master' into solid-mechanics-with-mesh-modules

josep-m-carbonell committed:
error in a update and merge of a file in swiming dem

josep-m-carbonell committed:
Merge branch 'feature-umat-constitutive' into solid-mechanics-with-mesh-modules

josep-m-carbonell committed:
Merge branch 'feature-u-w-element' into solid-mechanics-with-mesh-modules

josep-m-carbonell committed:
corrections to write section properties

josep-m-carbonell committed:
correction of beam sections

josep-m-carbonell committed:
adding Dofs in solvers input  json

josep-m-carbonell committed:
corrections to check input dofs

josep-m-carbonell committed:
Merge branch 'master' into solid-mechanics-with-mesh-modules

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
Merge branch 'solid-mechanics-with-mesh-modules' into develop

josep-m-carbonell committed:
Merge branch 'develop' into solid-mechanics-with-mesh-modules

josep-m-carbonell committed:
Removed submodule applications/PfemApplication/external_modules/triangle

josep-m-carbonell committed:
compile with triangle in external libraries

josep-m-carbonell committed:
Merge branch 'solid-mechanics-with-mesh-modules' into develop

josep-m-carbonell committed:
Merge branch 'master' into develop

josep-m-carbonell committed:
restore previous const_law_umat_app

josep-m-carbonell committed:
restore umat file

josep-m-carbonell committed:
rename models Umat to umat

josep-m-carbonell committed:
some overrides and name corrections

josep-m-carbonell committed:
Removed submodule applications/PfemApplication/external_modules/triangle

josep-m-carbonell committed:
Merge branch 'feature-g-pfem-develop' into develop

loumalouomega commented on kratos/solving_strategies/schemes/residual_based_bossak_displacement_scheme.hpp:
```diff
@@ -281,44 +281,44 @@ class ResidualBasedBossakDisplacementScheme: public Scheme<TSparseSpace,TDenseSp
             if (itNode -> IsFixed(ACCELERATION_X))
             {
-                CurrentDisplacement[0] = PreviousDisplacement[0] + DeltaTime * PreviousVelocity[0] + std::pow(DeltaTime, 2) * ( 0.5 * (1.0 -  2.0 * mNewmark.beta) * PreviousAcceleration[0] + mNewmark.beta * CurrentAcceleration[0]);
+                CurrentDisplacement[0] = PreviousDisplacement[0] + DeltaTime * PreviousVelocity[0] + DeltaTime * DeltaTime * ( 0.5 * (1.0 -  2.0 * mNewmark.beta) * PreviousAcceleration[0] + mNewmark.beta * CurrentAcceleration[0]);
```
Actually is better to use std::pow(val, 2)

loumalouomega commented on applications/ConstitutiveModelsApplication/custom_models/constitutive_model_data.hpp:
```diff
@@ -0,0 +1,692 @@
+      //Constructor
+      VariableValueData()
+      {
+	mpIntVariable    = NULL;
```
I recommend to use nullptr https://stackoverflow.com/questions/20509734/null-vs-nullptr-why-was-it-replaced

Can be replaced using clang tidy https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-nullptr.html

https://github.com/KratosMultiphysics/Kratos/wiki/How-to-use-Clang-Tidy-to-automatically-correct-code

loumalouomega commented on applications/ConstitutiveModelsApplication/custom_models/elasticity_models/borja_model.cpp:
```diff
@@ -0,0 +1,203 @@
+      KRATOS_TRY
+
+      rVolumetric = 0;
+      for (unsigned int i = 0; i < 3; i++)
```
It is a good practice to use ++i instead of i++ https://stackoverflow.com/questions/24853/what-is-the-difference-between-i-and-i

loumalouomega commented on applications/ConstitutiveModelsApplication/custom_models/elasticity_models/hyperelastic_model.cpp:
```diff
@@ -0,0 +1,1482 @@
+
+      StressMatrix *= 2.0;
+
+      rStressMatrix += StressMatrix;
```
noalias?

loumalouomega commented on applications/SolidMechanicsApplication/custom_elements/beam_elements/beam_element.cpp:
```diff
@@ -0,0 +1,1709 @@
+
+	rElementalDofList.push_back(GetGeometry()[i].pGetDof(DISPLACEMENT_X));
+	rElementalDofList.push_back(GetGeometry()[i].pGetDof(DISPLACEMENT_Y));
+	rElementalDofList.push_back(GetGeometry()[i].pGetDof(DISPLACEMENT_Z));
```
do you add always 3 displacements even in 2d?

loumalouomega commented on applications/SolidMechanicsApplication/custom_elements/beam_elements/beam_element.cpp:
```diff
@@ -0,0 +1,1709 @@
+  void BeamElement::CalculateMassMatrix(MatrixType& rMassMatrix, ProcessInfo& rCurrentProcessInfo)
+  {
+    KRATOS_TRY
+
```
Should this at least return a zero matrix?

josep-m-carbonell commented:
@loumalouomega. Stop you comments please and focus in your affairs.

loumalouomega commented on applications/SolidMechanicsApplication/custom_elements/beam_elements/beam_element.cpp:
```diff
@@ -0,0 +1,1709 @@
+
+    KRATOS_TRY
+
+
```
Should this at least return an zero value?

loumalouomega commented on applications/SolidMechanicsApplication/custom_processes/add_dofs_process.h:
```diff
@@ -292,7 +292,7 @@ class AddDofsProcess : public Process
 	int number_of_nodes = mr_model_part.NumberOfNodes();
 	ModelPart::NodeConstantIterator nodes_begin = mr_model_part.NodesBegin();
-	
+
```
The variables_utils can replace this process?

loumalouomega commented on applications/SolidMechanicsApplication/custom_processes/assign_scalar_field_to_conditions_process.h:
```diff
@@ -54,7 +54,6 @@ class AssignScalarFieldToConditionsProcess : public Process
         Parameters default_parameters( R"(
             {
                 "model_part_name":"MODEL_PART_NAME",
```
can this process be replaced by the one in the core?

loumalouomega commented on applications/SolidMechanicsApplication/custom_processes/assign_scalar_field_to_nodes_process.h:
```diff
@@ -54,7 +54,6 @@ class AssignScalarFieldToNodesProcess : public Process
             {
                 "model_part_name":"MODEL_PART_NAME",
-                "mesh_id": 0,
                 "variable_name": "VARIABLE_NAME",
```
idem

loumalouomega commented on applications/SolidMechanicsApplication/custom_processes/assign_scalar_variable_to_conditions_process.h:
```diff
@@ -43,15 +43,13 @@ class AssignScalarVariableToConditionsProcess : public Process
     AssignScalarVariableToConditionsProcess(ModelPart& model_part,
-						 Parameters rParameters
-						 ) : Process(Flags()) , mr_model_part(model_part)
+					    Parameters rParameters) : Process(Flags()) , mr_model_part(model_part)
```
idem

loumalouomega commented on applications/SolidMechanicsApplication/custom_processes/assign_scalar_variable_to_nodes_process.h:
```diff
@@ -43,15 +43,13 @@ class AssignScalarVariableToNodesProcess : public Process
     AssignScalarVariableToNodesProcess(ModelPart& model_part,
-				       Parameters rParameters
-				       ) : Process(Flags()) , mr_model_part(model_part)
+				       Parameters rParameters) : Process(Flags()) , mr_model_part(model_part)
```
idem

loumalouomega commented on applications/SolidMechanicsApplication/custom_processes/assign_vector_field_to_conditions_process.h:
```diff
@@ -64,7 +63,6 @@ class AssignVectorFieldToConditionsProcess : public Process
         // Validate against defaults -- this ensures no type mismatch
```
idem

loumalouomega commented on applications/SolidMechanicsApplication/custom_processes/assign_vector_variable_to_conditions_process.h:
```diff
@@ -43,15 +43,13 @@ class AssignVectorVariableToConditionsProcess : public Process
     AssignVectorVariableToConditionsProcess(ModelPart& model_part,
-					    Parameters rParameters
-					    ) : Process(Flags()) , mr_model_part(model_part)
+					    Parameters rParameters) : Process(Flags()) , mr_model_part(model_part)
```
idem

loumalouomega commented on applications/SolidMechanicsApplication/custom_processes/fix_scalar_dof_process.h:
```diff
@@ -101,9 +99,7 @@ class FixScalarDofProcess : public Process
-			const Variable<double>& rVariable,
-			std::size_t mesh_id
-			) : Process(), mr_model_part(model_part), mmesh_id(mesh_id)
+			const Variable<double>& rVariable) : Process(), mr_model_part(model_part)
```
can variables_utils replace this process?

loumalouomega commented on applications/SolidMechanicsApplication/custom_processes/free_scalar_dof_process.h:
```diff
@@ -43,23 +43,20 @@ class FreeScalarDofProcess : public Process
     ///@name Life Cycle
     ///@{
     FreeScalarDofProcess(ModelPart& model_part,
```
can variables_utils replace this process?

loumalouomega commented on applications/SolidMechanicsApplication/python_scripts/assign_modulus_and_direction_to_conditions_process.py:
```diff
@@ -30,7 +30,6 @@ def __init__(self, Model, custom_settings ):
         {
              "help" : "This process assigns a modulus and direction value to a conditions vector variable",
              "model_part_name": "MODEL_PART_NAME",
```
can be replaced by the one in the core?

loumalouomega commented on applications/SolidMechanicsApplication/python_scripts/assign_modulus_and_direction_to_nodes_process.py:
```diff
@@ -25,7 +25,6 @@ def __init__(self, Model, custom_settings ):
         {
              "help" : "This process assigns a modulus and direction value to a vector variable",
              "model_part_name": "MODEL_PART_NAME",
```
can be replaced by the one in the core?

loumalouomega commented on applications/SolidMechanicsApplication/python_scripts/assign_scalar_to_nodes_process.py:
```diff
@@ -32,7 +32,6 @@ def __init__(self, Model, custom_settings ):
              "help" : "This process assigns a scalar value to a scalar variable",
              "model_part_name": "MODEL_PART_NAME",
-             "mesh_id": 0,
              "variable_name": "VARIABLE_NAME",
```
can be replaced by the one in the core?

loumalouomega commented:
I added some commentaries, but certainly there is a lot of code to review

loumalouomega commented:
I was trying to help

josep-m-carbonell commented:
@loumalouomega. Thank you very much. You have done more than expected. Taking into account that nobody asked you as a reviewer. Even more, after doing parallel developmemts and copying code into another application in order to not contribute in SolidMechanics. Yourself can guess that you comments and opinions here are not appropiate.

roigcarlo commented on .gitmodules:
```diff
@@ -0,0 +1,3 @@
+[submodule "applications/PfemApplication/external_modules/tetgen"]
+	path = applications/PfemApplication/external_modules/tetgen
+	url = https://github.com/PFEM/tetgen.git
```
Not necessarily wrong but just some days ago we had to remove tetgen from the repo (see #906) for licencing issues.

To my understanding it now works as a external library like metis, trilinos etc... I honestly don't know if adding it as a submodule even if the code is hosted in your own repo ( which I understand it hold the original licence of tetgen ) is compatible with BSD.

josep-m-carbonell replied:
Tetgen was not removed from the PfemApplication. Here the change is to replace it by a submodule. Therefore it is not in the repo anymore.

lluis-mv reviewed (approved):
it is what we have been working on the fork and to me it is OK

roigcarlo reviewed (commented):
Having a first look at the PR I have some concerns. I will let it compile to check everything is ok.

As a personal note, I understand that adding applications will generate huge PR's by definition but I would prefer to keep it as small as possible ( maybe 4 different PR, 1 per app?). It is far easier for everyone to look at the changes.

Also, reviews are open to **everyone**.

roigcarlo commented on applications/UmatApplication/CMakeLists.txt:
```diff
@@ -0,0 +1,60 @@
+# define library Kratos which defines the basic python interface
+add_library(KratosUmatApplication SHARED ${KRATOS_UMAT_APPLICATION_SOURCES})
+target_link_libraries(KratosUmatApplication KratosCore KratosConstitutiveModelsApplication)
+set_target_properties(KratosUmatApplication PROPERTIES PREFIX "")
```
UMAT seems to be using `KRATOS_API(UMAT_APPLICATION)` macro but `UMAT_APPLICATION` in not defined here. This will cause it will never export in windows. Don't know if its intended

josep-m-carbonell replied:
You explained to me that there was some problems in fortran compilation for  windows. If they are solved I will thank you if you add the proper corrections.

lluis-mv replied:
Hi, it is sort of intended. In this application we are using Fortran code, we thought it will be easier to compile in Linux and that's why we created this tiny application. Of course, this is almost a draft of the feature (be able to incorporate constitutive laws written in the Umat format, typical of comercial FEM softwares) that works but some more work is required to make it relevant.

roigcarlo replied:
Well is not a problem per-se. It works with intel we just don't have the licence so we cannot use it, apart from that its just fine. 

In my opinion, it doesn't hurt to add the define as you done the hard part in the code, but I understand what you mean.

lorenzogracia commented:
I'm going to check the pull request in an external branch in order to verify if Dam and Poromechanics can be affected in some way.

roigcarlo commented on applications/ConstitutiveModelsApplication/tests/generalTests.py:
```diff
@@ -0,0 +1,28 @@
+    return os.path.dirname(__file__) + "/" + fileName
+
+
+class KratosConstitutiveModelsGeneralTests(KratosUnittest.TestCase):
```
This test seems to do nothing. No need to include it just because its need to have one, If you intend to add tests latter it is ok.

josep-m-carbonell replied:
We are buiding the tests, they are not included there.

roigcarlo commented on applications/ConstitutiveModelsApplication/tests/test_ConstitutiveModelsApplication.py:
```diff
@@ -0,0 +1,53 @@
+from generalTests import KratosConstitutiveModelsGeneralTests
+
+
+def AssambleTestSuites():
```
Same as above

AFranci commented on applications/PfemFluidDynamicsApplication/custom_processes/adaptive_time_interval_process.hpp:
```diff
@@ -284,148 +284,160 @@ namespace Kratos
 
 
+	      }
 	    }
```
Why do you want to modify this file? Are these the same changes that crashed the code some days ago?

josep-m-carbonell replied:
I hope the merge is done correctly. The changes were done to allow to print walls in GiD output. It was interferring in this process and this correction was needed.

AFranci replied:
This function is for reducing adaptively the time step, I do not see how it can interfere with the GiD output. Anyway, there are changes of the function aim that I do not understand. Please, let the previous version of the function and, if required, we will modify in a new specific pull request.

josep-m-carbonell replied:
Please try to comment the new lines and test the behaviour. Then you will understand why they are needed.

AFranci replied:
we will discuss it the next week in the office, now I cannot test everything...

josep-m-carbonell replied:
To keep the same behaviour of the process wall elements must not be considered in the loop. This new elements are there for writing purposes. The original implementation is not aware of this matter and without this change the process fails.

lluis-mv commented on applications/PfemApplication/custom_bounding/spatial_bounding_box.hpp:
```diff
@@ -852,39 +860,35 @@ class KRATOS_API(PFEM_APPLICATION) SpatialBoundingBox
 	start=1;
```
Hi, I think these four lines (from unsigned int start = 0 to start=1) may be removed since the variable start is not used and creates a compiler warning. KR

josep-m-carbonell replied:
It is possible mistake. I have not detected it. Please can you add the correction to the branch?

lluis-mv replied:
I think I have done it ;-P

AFranci commented on applications/PfemFluidDynamicsApplication/custom_processes/model_start_end_meshing_for_fluids_process.hpp:
```diff
@@ -403,38 +411,63 @@ namespace Kratos
+      // rComputingModelPart.Conditions().Unique();
       
       if( EchoLevel > 1 )
 	std::cout<<"    [ SUBMODEL PART ["<<rComputingModelPart.Name()<<"] [Elems="<<rComputingModelPart.NumberOfElements()<<"|Nodes="<<rComputingModelPart.NumberOfNodes()<<"|Conds="<<rComputingModelPart.NumberOfConditions()<<"] ] "<<std::endl;
```
Did you test if the pfem fluid still works after these modifications? If yes, for me it is ok

josep-m-carbonell replied:
There are no tests in the application. I have tested some cases but I can not ensure all cases.

AFranci reviewed (commented):
Apart from the two highlighted concerns, the rest of the changes on the Pfem Fluid are fine

josep-m-carbonell commented:
Thank you @AFranci,  I will apreciate if you can compile it and check some of the classical cases.

lluis-mv committed:
removing 4 lines of code that were not used

pooyan-dadvand commented:
@loumalouomega Thanks for your review. Actually I agree with many of them. 

@josep-m-carbonell would you please break this pull request in some smaller ones ? The pull request is too large to be reviewed. (Is even difficult to see if the changes affect core or other applications)

pooyan-dadvand reviewed (changes requested):
I would just block it to avoid accidental merge.... so @josep-m-carbonell can divide it in smaller PRs.

roigcarlo commented on kratos/includes/variables.h:
```diff
@@ -47,6 +47,7 @@ namespace Kratos
 
     //ints
     KRATOS_DEFINE_VARIABLE( int, DOMAIN_SIZE )
+    KRATOS_DEFINE_VARIABLE( int, DIMENSION )
```
I strongly suggest giving this var another name as we have plenty of non-kratos variables and enums named `DIMENSION` in other apps.

josep-m-carbonell replied:
Well this is a  Kratos Variable used for other purposes. If there is any conflict in the future it can be changed.

roigcarlo replied:
The problem is that if we left it this way it does conflict, just the fact that it exists is a problem. For example, in DEM application there are multiple enums that define DIMENSION as a integer ( this one translates as a KratosVariable<int>). And probably it is not a easy to detect problem when/if it happens. 

Specifically what I amb afraid of is that some kind of partial shadowing could happend depending on which files are included or not. For example consider this code:

```C++
#include "variables.h"
#include "myapp.h" #but you don't include the file with the enum {DIMENSION = 3}

for(int i = 0; i < DIMENSION; i++) {
    doSomething();
}
```

This code probably compiles, and maybe only loop once, maybe it crashes, maybe loops 4 times and we end up geting a memleak.

Another solution would be to move it to your application or to pfem base / contact emchanichs if it needs to be inherited by anyone else. This way is isolated from other people, then the problem is only while coupling with other applications, which is perfectly fine.

josep-m-carbonell replied:
Ok I see, the problem is critical and I have new questions:
1) This happens because these enums write the variables in CAPITALS which coincide with Kratos Variables standard ?  If the enum name is "Dimension" then no problem occurs?
(if it is like this, would be convienient to not use capitals for enum variables ?)
2) If a Kratos Variable and a Global or Local flag have the same name (they are usually written in capitals), have we got the same problem?
3) I think it is planned (in the future) to define the variables in the applications, not in Kratos core, then variables could be repeated. However it will be also dangerous because we can find the same variable defined as a local variable inside some classes.
I think you must expose this problem to @KratosMultiphysics/technical-committee .

Right now I will change the name to SPACE_DIMENSION to avoid conflicts and I will add the correction to this branch.

roigcarlo replied:
1) Yes, that is the problem indeed. As you said it won't be a problem if DIMENSION enum or C++ variables would have been called "Dimension" without caps, but enums usually are written in caps to show that are meaningful values.
2) Yes, if you are aware of any flag where this happens pls tell me.
3) Again, you are right. While in general it should not be the case this may happen. Probably the problem with dimension is that is somehow a very "generic" word. I recall having similar issues with VELOCITY for example. We will for sure discuss this in the next @KratosMultiphysics/technical-committee meeting.

josep-m-carbonell committed:
missing space in script

josep-m-carbonell committed:
Merge branch 'master' into feature-gpfem-constitutive-develop

josep-m-carbonell committed:
Merge branch 'master' into feature-gpfem-constitutive-develop

josep-m-carbonell committed:
restore pow()  

It is worse but  for human understading of the code and maybe less optimal. But it is better to not start a stupid discussion.

josep-m-carbonell committed:
restore format with the extra spaces

lorenzogracia commented:
Hi @josep-m-carbonell , I have created a branch in which I have merged your pull request with my branch (it is updated with the master), everything goes fine except a compilation error in the scheme ExplicitCentralDifferenesScheme where appears an error in pragma omp. I attach a snapshot. 

![imagen](https://user-images.githubusercontent.com/25606813/32389573-5b10c766-c0cb-11e7-80fc-65406d3306dd.png)

Can u check?

Regards

josep-m-carbonell commented:
Hi, @lorenzogracia, I know about that, but it does not happen to me, I think it is something related to the compiler in your ubuntu distribution, because the line seems correct. Maybe @roigcarlo can tell us a solution for that.

I will set mTime as private, not firstprivate. After the merge of the changes try to compile it again please.

josep-m-carbonell commented:
@pooyan-dadvand , changes in the core included in this pull request are added separately in another pull request #963. So you will be able to check them. Other changes concern only to applications.

josep-m-carbonell committed:
change DIMENSION to SPACE_DIMENSION

josep-m-carbonell committed:
change DIMENSION to SPATIAL_DIMENSION in applications

josep-m-carbonell committed:
correct omp first private compilation error

and compilation warning unsigned int

josep-m-carbonell committed:
improvements in zeromatrix initializations

josep-m-carbonell committed:
Merge branch 'master' into feature-gpfem-constitutive-develop

josep-m-carbonell committed:
delete a print in python

josep-m-carbonell committed:
Merge branch 'feature-#954-changes-core' into feature-gpfem-constitutive-develop

josep-m-carbonell commented:
Pull request of the core #963 was reviewed, corrected, approved and merged. Once the code compiles for @lorenzogracia. Please @pooyan-dadvand  unblock it.

lorenzogracia commented:
Hi @josep-m-carbonell, still not compiles... I don't know why. @roigcarlo any idea of that?

![imagen](https://user-images.githubusercontent.com/25606813/32417371-ecfa0a0a-c258-11e7-876d-042ca2e88a79.png)

roigcarlo commented:
Yep, let me take a closer look when I arrive home but the error states that
mTime is not defined

On Sun, Nov 5, 2017, 6:42 PM Lorenzo Gracia <notifications@github.com>
wrote:

> Hi @josep-m-carbonell <https://github.com/josep-m-carbonell>, still not
> compiles... I don't know why. @roigcarlo <https://github.com/roigcarlo>
> any idea of that?
>
> [image: imagen]
> <https://user-images.githubusercontent.com/25606813/32417371-ecfa0a0a-c258-11e7-876d-042ca2e88a79.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/KratosMultiphysics/Kratos/pull/954#issuecomment-341991041>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB2Jr0jLgplqMdRjJy-smD7j5CXLKJAXks5szfOUgaJpZM4QQUSI>
> .
>

roigcarlo commented:
@lorenzogracia The problem here is that mTime is a struct, wich can only be private under certain circumstances. In this case `TimeVariables` needs to have a copy constructor, which is not explicitly declared (compiler will implicitly declare one) maybe the problem is somehow here.
See [This Link, section 25](https://software.intel.com/en-us/articles/32-openmp-traps-for-c-developers#IDAXSRDC) for more info.

Anyway, in the screen code there is no reason to consider mTime private as it is only read in the loop, so I would suggest to remove it if is giving problems. But the code seems fine...

josep-m-carbonell commented:
Thank you @roigcarlo. Please @lorenzogracia , can you try compile without 'private(mTime)' and see if it works?  I have eliminated it.

josep-m-carbonell committed:
deleting mTime private

lorenzogracia commented:
Thank you @roigcarlo. Now it works.

josep-m-carbonell committed:
load reset when interval finishes

josep-m-carbonell committed:
Merge branch 'master' into feature-gpfem-constitutive-develop

josep-m-carbonell committed:
restore ValidateInput in constitutive_law

josep-m-carbonell committed:
python print deleted

pooyan-dadvand commented:
Seems that it is working for @lorenzogracia and rest of the team. Only @maceligueta is has not confirmed. Meanwhile I unblock it to start making release branch.
